### PR TITLE
fix(api-extractor): Don't mark items documented with {@inheritDoc} references to package-external items as "undocumented" 

### PR DESCRIPTION
**Pull request by Josmithr**
`{@inheritDoc}` references to items outside of a given package cannot currently be resolved / evaluated by api-extractor (see #1195). The logic for `ae-undocumented` did not take this into account, and was treating any API item with an unresolvable inheritDoc comment as "undocumented". In reality, the "documented" status of the item cannot be determined by API-Extractor. To be safe, we no longer mark items in this ambiguous state as "undocumented".

Fix for #3462

Also adds a test scenario for `@inheritDoc`, and updates the repo's API reports to reflect the new policy.

### CODE REVIEW

File: apps/api-extractor/src/collector/ApiItemMetadata.ts [hunk 1]
```diff
@@ -75,7 +75,19 @@ export class ApiItemMetadata {
    */
   public tsdocComment: tsdoc.DocComment | undefined;
 
-  // Assigned by DocCommentEnhancer
+  /**
+   * Tracks whether or not the associated API item is "undocumented".
```
Comment (octogonz) [inline apps/api-extractor/src/collector/ApiItemMetadata.ts:79]: `Tracks whether or not the associated API item is known to be missing sufficient documentation.`

**Review by octogonz — APPROVED** [review]

File: apps/api-extractor/src/collector/ApiItemMetadata.ts (hunk 1 above)
Comment (Josmithr) [reply apps/api-extractor/src/collector/ApiItemMetadata.ts:79]:
```suggestion
   * Tracks whether or not the associated API item is known to be missing sufficient documentation.
```

**Comment by octogonz** [comment]
@Josmithr  thank you for investigating this and figuring out how to fix it! 🎉

**Comment by octogonz** [comment]
🚀  Released with API Extractor 7.42.0

**Pull request merged**